taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -2147,8 +2146,34 @@ void CodeGenLLVM::create_offload_struct_for(OffloadedStmt *stmt, bool spmd) {
             alloca_type->getArrayNumElements() == 1 &&
             alloca_type->getArrayElementType() == char_type) {
           auto new_type = llvm::ArrayType::get(char_type, stmt->tls_size);
-          alloca->setAllocatedType(new_type);
```

Comment:
Could we keep this and just setSourceElementType on all the GEP users instead of create new alloca and GEP?

Reply:
I tried and it raised error because the type of the AllocaInst is still [1 × i8].

Reply:
How about add alloca->mutateType(updatedPointerType)

Reply:
It raises "isa<X>(Val) && \"cast<Ty>() argument of incompatible type!\"" when cloning the AllocaInst when cloning the module. Also, the description of `mutateType` said "Note that this is an extremely dangerous operation which can create completely invalid IR very easily. It is strongly recommended that you recreate IR objects with the right types instead of mutating them in place." so I don't think it is a good idea to use this function...